### PR TITLE
Basic Delegate Trust #58

### PR DESCRIPTION
**Pull request by IanMendozaJaimes**
- Added functions to delegate and cancel delegation of trust
- Added tests
- Added new parameter `votedel.mul` to settings

### CODE REVIEW

File: include/seeds.proposals.hpp [hunk 1]
```diff
@@ -89,6 +89,12 @@ CONTRACT proposals : public contract {
 
       ACTION initnumprop();
 
+      ACTION delgatetrust(name delegator, name delegatee, name scope);
+
+      ACTION mimicvote(name delegatee, name scope, uint64_t proposal_id, double percentage_used, name option, uint64_t chunksize);
+
+      ACTION cncldeltrst(name delegator, name scope);
```
Comment (n13) [inline include/seeds.proposals.hpp:96]: delegate()
undelegate()

for these functions?

File: include/seeds.proposals.hpp [hunk 2]
```diff
@@ -242,6 +251,16 @@ CONTRACT proposals : public contract {
 
         uint64_t primary_key()const { return account.value; }
       };
+
+      TABLE delegate_trust_table { // scoped by proposal's category (alliance, campaign, etc)
+        name delegator;
+        name delegatee;
+        double weight;
```
Comment (n13) [inline include/seeds.proposals.hpp:258]: cool!

File: src/seeds.proposals.cpp [hunk 3]
```diff
@@ -795,8 +804,7 @@ bool proposals::revert_vote (name voter, uint64_t id) {
   return false;
 }
 
-void proposals::vote_aux (name voter, uint64_t id, uint64_t amount, name option, bool is_new) {
-  require_auth(voter);
+void proposals::vote_aux (name voter, uint64_t id, uint64_t amount, name option, bool is_new, bool is_mimic) {
```
Comment (n13) [inline src/seeds.proposals.cpp:807]: why "mimic" and not "delegate" ?

File: src/seeds.proposals.cpp [hunk 4]
```diff
@@ -1271,3 +1295,131 @@ ACTION proposals::initnumprop() {
...
+    ditr++;
+    count++;
+  }
+
+  if (ditr != deltrusts_by_delegatee.end() && ditr -> delegatee == delegatee) {
+    action next_execution(
+      permission_level{get_self(), "active"_n},
+      get_self(),
+      "mimicvote"_n,
+      std::make_tuple(delegatee, scope, proposal_id, percentage_used, option, chunksize)
```
Comment (n13) [inline src/seeds.proposals.cpp:1399]: Will this not loop forever when size > chunk size? 

The second call signature is the exact same as the first, no pointers are moving. That's a lot of votes..

Comment (n13) [reply src/seeds.proposals.cpp:1399]: Suggest to add a test with 2 delegations and chunk size 1

**Review by n13 — CHANGES_REQUESTED** [review]

File: include/seeds.proposals.hpp (hunk 1 above)
Comment (IanMendozaJaimes) [reply include/seeds.proposals.hpp]: I changed the names of the functions

File: include/seeds.proposals.hpp (hunk 2 above)
Comment (IanMendozaJaimes) [reply include/seeds.proposals.hpp:258]: jaja thanks, but I am not using the weight yet, I wasn't sure where I had to use it.

File: src/seeds.proposals.cpp (hunk 3 above)
Comment (IanMendozaJaimes) [reply src/seeds.proposals.cpp]: I changed the name of the variable

File: src/seeds.proposals.cpp (hunk 4 above)
Comment (IanMendozaJaimes) [reply src/seeds.proposals.cpp]: You are completely right, what a horrible mistake I made here. I fixed in the commit

File: include/seeds.proposals.hpp (hunk 2 above)
Comment (n13) [reply include/seeds.proposals.hpp:258]: It's cool weight is in there

**Review by n13 — APPROVED** [review]

**Pull request merged**